### ISSUE TITLE
Fedora Support

### ISSUE THREAD

**Issue by base-zero** [nginxinc/nginx-amplify-agent#26]
Would be nice to have Fedora support out of the box.

**Comment by dedm** [comment]
Soon we will provide a step-by-step manual "How to install NGINX Amplify Agent if your OS is not supported". Hope it will help.

**Comment by carlbennett** [comment]
I'd love for this feature as well! It'd be great if Fedora was supported _and_ documentation on how to install manually on unsupported OSes.

**Comment by dedm** [comment]
@carlbennett, @base-zero  we released a beta script for installing agent in non-supported systems: https://github.com/nginxinc/nginx-amplify-agent/blob/master/packages/install-source.sh 

Please try it and tell us how was it.

**Comment by carlbennett** [comment]
Fedora is non-supported? What will it take to support it? Since Fedora 16 and CentOS 7.0, they both are pretty much the same when it comes to installing packages. Why not duplicate that effort?

**Comment by base-zero** [comment]
I have used the script provided and installed it without any problems however no data is being sent to the dashboard.

**Comment by dedm** [comment]
@base-zero yeah, we found a bug that will be fixed in the next release (0.39). We're planning to release it on Sep 20 or 21. Please bear with us!

**Comment by dedm** [comment]
@carlbennett sorry, right now we are not able to build Fedora packages due to lack of resources. In 5-6 months we will think about Fedora again.

**Comment by base-zero** [comment]
@dedm Is there any news on Fedora support now ?

**Comment by ptreyes** [comment]
Hey, still no packages unfortunately. But this works on Fedora 24:

https://github.com/nginxinc/nginx-amplify-doc/blob/master/amplify-faq.md#21-what-operating-systems-are-supported
https://raw.githubusercontent.com/nginxinc/nginx-amplify-agent/master/packages/install-source.sh

Do you have any issues with the above script?

**Comment by base-zero** [comment]
@ptreyes Sorry maybe I should have been more specific with the Fedora version. I did get that script to work with Fedora 24 back in September but our new servers are now Fedora 26 and that script doesn't work.

**Comment by carlbennett** [comment]
I second this. I am currently on Fedora 25 and 26 in my systems. I also encountered the no data availability in the dashboard with the agent installed this way. It would be great if there was more support given to the Fedora operating system. It's just so much nicer having all the latest package updates than it is to be a few versions behind, and this monitoring agent would be an excellent compliment to the suite of services I use.

Right now, I use self-hosted Grafana and Prometheus for monitoring, since this agent doesn't work well for my use cases.

I am also quickly upgrading my infrastructure from Fedora 25 to 26. I have one more machine to decommission before that's complete.

**Comment by gshulegaard** [comment]
Thanks for the feedback guys!  It's always useful to hear what our users want to help us prioritize resources.

**Comment by ptreyes** [comment]
I've done some updates (+testing) to the script for Fedora 26. Can someone (@carlbennett, @base-zero) check it out? In my testing it seemed to work :) Would be happy to hear any feedback.

**Comment by base-zero** [comment]
@ptreyes Sorry for the delay, sure happy to try it. Just tried https://raw.githubusercontent.com/nginxinc/nginx-amplify-agent/master/packages/install-source.sh but it now fails with the following error ```4. Checking OS compatibility ... fedora is currently unsupported, apologies!```

**Comment by base-zero** [comment]
@ptreyes Sorry for the slow reply. I have install it on Fedora 26 now and the install has worked. Will leave it for a few day and se how it gets on !

**Comment by ptrunk** [comment]
I am trying to build nginx-amplify-agent from source for Fedora 27 but i it always fails with
```
+ /usr/lib/rpm/find-debuginfo.sh -j4 --strict-build-id -m -i --build-id-seed 0.47-1.fc27 --unique-debug-suffix -0.47-1.fc27.x86_64 --unique-debug-src-base nginx-amplify-agent-0.47-1.fc27.x86_64 --run-dwz --dwz-low-mem-die-limit 10000000 --dwz-max-die-limit 110000000 -S debugsourcefiles.list /home/paul/rpmbuild/BUILD/nginx-amplify-agent-0.47-1
find: 'debug': No such file or directory

[...]

error: Empty %files file /home/paul/rpmbuild/BUILD/nginx-amplify-agent-0.47-1/debugsourcefiles.list
```
Any ideas or hints? Thank you :-)

Full build log:
[build.log](https://github.com/nginxinc/nginx-amplify-agent/files/1490562/build.log)

**Comment by ptreyes** [comment]
Hi,

I'll test it and let you know :)

**Comment by ptrunk** [comment]
Thank you very much for your help!

My workaround is now to add
```
%define  debug_package %{nil}
```
to my .spec file (i have used the one which is in packages/rpm). For me it is not necessary to have an extra debug package but i think it would be helpful if the spec file works without such custom changes :-).
